api/src/main/java/jakarta/enterprise/inject/spi/CDI.java

```diff
@@ -167,4 +169,14 @@ public BeanContainer getBeanContainer() {
         return getBeanManager();
     }
 
+    /**
+     * Get the Build Compatible Extension {@link AnnotationBuilderFactory} for the current CDI container.
+     * <p>
+     * This is present here to reuse existing service provider discovery mechanism.
+     * Should not be called directly by users; the static methods on {@link AnnotationBuilder} are preferred.
+     *
+     * @return the {@link AnnotationBuilderFactory}
+     */
+    // TODO this is easiest, but also exposes AnnotationBuilderFactory prominently, which we don't want
+    public abstract AnnotationBuilderFactory getAnnotationBuilderFactory();
```

Comment:
Perhaps just with service loader inside the `AnnotationBuilderFactory` interface it self:

```java
interface AnnotationBuilderFactory {
      AnnotationBuilderFactory INSTANCE = SomePackagePrivateClass.resolveFactory()
      ...
```

Reply:
Yea, I just didn't want to reimplement all the service discovery stuff with priorities -- which we probably would have to do, since the `CDI` class does ti too.

Reply:
I mean -- yea, I'll probably just do it, it's just copy'n'paste :-)

Reply:
Done that. I'm now thinking if `AnnotationBuilder` should perhaps belong to `jakarta.enterprise.lang.model`, since it's fully self-contained. Probably don't have to solve this in this PR, can move later if we find that makes more sense.